src/Condition.js

```diff
@@ -32,27 +32,28 @@ class Conditional {
           : (...values) => values.every(value => value === is);
 
       this.fn = function(...values) {
-        let currentSchema = values.pop();
+        let options = values.pop();
+        let schema = values.pop();
         let option = isFn(...values) ? then : otherwise;
 
-        return option(currentSchema);
+        return option(schema, options);
```

Comment:
why the API change here?

Reply:
First argument is the same. Additional second argument is used in this:

```js
return (base, options) => base.concat(schema.resolve(options));
```

So that `Lazy` can be `concat`ed.

Reply:
I see, let's inline the `callOrConcat` function here, it's not adding any value separate. I'm incline to say we should keep the public api the same for when it's a function...

Reply:
Did rebase on current master and did refactor Condition.js file. Just split logic into separate functions, no functional changes except requested removal of `options` object from custom functions api.